### PR TITLE
Composer changes

### PR DESCRIPTION
**Pull request by jeremiahsmall**
Add composer.lock and remove composer.phar per this discussion: https://github.com/zfcampus/zf-apigility-skeleton/pull/92#issuecomment-109941631
(Moving the Vagrant addition to its own PR)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,24 +13,37 @@ Installation
 Using Composer (recommended)
 ----------------------------
 The recommended way to get a working copy of this project is to clone the repository
-and use `composer` to install dependencies using the `create-project` command:
+and use `composer` to install dependencies using the `install` command:
 
-    curl -s https://getcomposer.org/installer | php --
-    php composer.phar create-project -sdev --repository-url="https://packages.zendframework.com" zendframework/skeleton-application path/to/install
+```
+git clone https://github.com/zfcampus/zf-apigility-skeleton.git # optionally, specify the directory in which to clone
```
Comment (samsonasik) [inline README.md]: zf-apigility?

**Comment by jeremiahsmall** [comment]
@samsonasik thanks for spotting the copy/paste issue with the github url in the readme.

**Comment by RobQuistNL** [comment]
:+1: the proper way.

**Comment by gianarb** [comment]
:+1: for composer.lock..
Composer phar into the VCS was added for an old discussion and for help a speedy project startup..

I agree with you but this implementation has already discussed

**Comment by RobQuistNL** [comment]
@gianarb whats so speedy about it? Its outdated, you even have to run self-update before running it. This downloads a new phar, just like the updated description does. If any, this is faster, because the user won't have to download the outdated .phar at all :)

**Comment by gianarb** [comment]
composer.phar is outdated is because this project has the problem of maintenance..
obviously it borns to be always updated

**Comment by RobQuistNL** [comment]
Then why does the README.md tells you to run "self-update"? Composer is very well maintained and updates to a propery release pretty often, so the self-update would download a new version very often. Honestly, I think the composer.phar in the project adds a little confusion for new users. Especially since it will show up as a change after the self-update.

**Comment by gianarb** [comment]
Oh yes I agree with you composer.phar into the VCS creates confusion and is
a bad practice IMO :) I hope that this issue resolve
this misunderstanding..
But this argument is already discussed here
https://github.com/zendframework/ZendSkeletonApplication/issues/62


2015-06-09 10:22 GMT+02:00 Rob Quist <notifications@github.com>:

> Then why does the README.md tells you to run "self-update"? Composer is
> very well maintained and updates to a propery release pretty often, so the
> self-update would download a new version very often. Honestly, I think the
> composer.phar in the project adds a little confusion for new users.
> Especially since it will show up as a change after the self-update.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/ZendSkeletonApplication/pull/286#issuecomment-110272778>
> .
>



-- 
Gianluca Arbezzano
www.gianarb.it

**Comment by akrabat** [comment]
I'm +1 for removing composer.phar. It made sense back in 2012, but doesn't any more. 

Ping @weierophinney for his thoughts as he had the last word on #62.

**Comment by ezimuel** [comment]
I discussed with @weierophinney and we both agree, we are going to remove `composer.phar` in the skeleton applications now and for the future. Having a correct README that shows how to install composer is the correct solution.

**Comment by gianarb** [comment]
:+1: Beautiful!

**Comment by akrabat** [comment]
@jeremiahsmall In the README, please put back the `create-project` method as this is the best way to install it. (Note, rebase from master first as I've just merged an improvement to the create-project command.)

Then remove the part about cloning manually and then running composer as there's no point in doing it this way given that we now expect the user have a composer installed.

**Comment by gianarb** [comment]
@ezimuel have you discussed about the vendor this?  
Maybe we can remove even this directory

**Comment by jeremiahsmall** [comment]
I've updated the README to include the `create-project` as Option 1, however, I've also shown how to do the same without curl as Option 2, and I've added steps to copy your newly minted composer.phar into your project afterward. I've also included two additional options. Option 3 is the via curl and tar option that was already in the README, and Option 4 is manual git clone and non-curl composer installation.

For your reading convenience: https://github.com/jeremiahsmall/ZendSkeletonApplication/blob/hotfix/composer-changes/README.md

Frankly, installing composer.phar first outside the project and then using it for `create-project` seems a bit more cumbersome to me, because it then either gets abandoned in some arbitrary directory, and composer still has to be installed inside the project, or it has to be moved into the project (as I've suggestion in options 1 & 2).

**Comment by akrabat** [comment]
I'm curious. Why do you think that composer has to be installed inside the project? It's perfectly happy living in /usr/local/bin.

**Comment by jeremiahsmall** [comment]
@akrabat, it's not that it *has* to be installed in the project, or anywhere in particular for that matter. It's just that in the "preferred" recipe it suggests you run `curl -s https://getcomposer.org/installer | php` which simply downloads the phar anywhere you happen to be. I think it's out of scope for a README like this to try to explain best practices for how/where to install composer.

Based on that, and on the fact that we can't count on curl, I'm actually in favor of only showing the Option 4 recipe.

**Comment by akrabat** [comment]
Merged after rebasing.

**Comment by jeremiahsmall** [comment]
Thx. Small typo in the word "donload" on revised README.md

**Comment by jeremiahsmall** [comment]
Also, still not clear why we put curl in the default instructions when there is another equally simple alternative with readfile.

**Comment by akrabat** [comment]
Fixed Typo. I picked curl as that's the first option in the Composer documentation and was already in the README. One way as an example is enough with a link to the documentation. We don't document using Composer-Setup.exe either, or installation on an IBM i-series, so we don't need to cover all bases.

**Pull request closed without merging**